### PR TITLE
Add match entries for a bunch of coming cs42l43 laptop SKUs

### PR DESCRIPTION
**Pull request by charleskeepax**

### CODE REVIEW

**Review by plbossart — DISMISSED** [review]
sounds good but don't we also need the machine driver quirks?

**Comment by bardliao** [comment]
> sounds good but don't we also need the machine driver quirks?

@plbossart I think quirks is needed only if cs42l43 speaker is used or cs42l43 works as bridge mode.

**Review by bardliao — DISMISSED** [review]

**Comment by charleskeepax** [comment]
Yeah some quirks will need to be added, but I am being a little more cautious with sending up the quirks as they depend on the SSID and as we found out with the last set sometimes those aren't as locked down as I thought. Where as these bus layouts are locked down pretty early.

File: sound/soc/intel/common/soc-acpi-intel-arl-match.c [hunk 1]
```diff
@@ -15,6 +15,112 @@ static const struct snd_soc_acpi_endpoint single_endpoint = {
...
+
+static const struct snd_soc_acpi_adr_device cs35l56_3_l1_adr[] = {
+	{
+		.adr = 0x00033101fa355601ull,
+		.num_endpoints = 1,
+		.endpoints = &spk_l_endpoint,
+		.name_prefix = "AMP1"
+	},
+};
+
```
Comment (charleskeepax) [inline sound/soc/intel/common/soc-acpi-intel-arl-match.c:123]: Lets hold off on merging this for a few more days, some questions on email about if these two adr's are correct.

Comment (charleskeepax) [reply sound/soc/intel/common/soc-acpi-intel-arl-match.c:123]: Ok small update pushed to the IDs should be good to rock now.

**Review by bardliao — APPROVED** [review]

**Review by plbossart — APPROVED** [review]

**Comment by plbossart** [comment]
@charleskeepax @bardliao are all the topologies listed in this PR part of the SOF tree now?

Confused about this PR https://github.com/thesofproject/sof/pull/9355 which is still a draft ....

**Comment by bardliao** [comment]
> @charleskeepax @bardliao are all the topologies listed in this PR part of the SOF tree now?
> 
> Confused about this PR [thesofproject/sof#9355](https://github.com/thesofproject/sof/pull/9355) which is still a draft ....

This is a draft because it has not been tested on the actual device, yet.

**Comment by charleskeepax** [comment]
Most of these are missing actually topologies yet, do you generally prefer to merge the topologies before the match entries? I could probably put a patch together although I don't have hardware to test most of them.

**Comment by plbossart** [comment]
it's usually better to merge and send this stuff upstream due to the development process, the patches will not be in Linus' tree before end of the year. 
But we do want to avoid misses in topology and make sure the sof-bin stuff does have the topologies ready. This is best tracked with a Sof GitHub/topology issue and a milestone.

**Pull request merged**